configuration/jsr223.md

```diff
@@ -0,0 +1,210 @@
+---
+layout: documentation
+title: JSR223 Scripting
+---
+
+{% include base.html %}
+
+# JSR223 Scripting
+> Note: This feature is for advanced users who have or are willing to learn some programming skills and are comfortable working with the command line prompt of the operating system hosting openHAB.
+
+## Overview
+
+[JSR223](https://docs.oracle.com/javase/6/docs/technotes/guides/scripting/) ([spec](https://jcp.org/aboutJava/communityprocess/pr/jsr223/index.html)) is a standard scripting API for Java Virtual Machine (JVM) [languages](https://en.wikipedia.org/wiki/List_of_JVM_languages). The JVM languages provide varying levels of support for the JSR223 API and interoperability with the Java runtime. Currently the following languages are know to work well for openHAB scripting.
+
+* [Oracle Nashorn](http://www.oracle.com/technetwork/articles/java/jf14-nashorn-2126515.html) (Javascript bundled with Java)
+* [Jython](http://www.jython.org/)
+* [Apache Groovy](http://www.groovy-lang.org/)
+
+Although JSR223 scripts can be used as a general-purpose extension language for openHAB, the most common usage will be for defining openHAB rules.
+
+### Language-Specific Documentation
+
+* [Javascript](jsr223-js.html)
+* [Jython](jsr223-jython.html)
+* Groovy (TODO)
+
+### Script Locations
+
+Scripts should be placed in the 'USERDATA/automation/jsr223' directory. For example, in Linux installations created with a
+package installer, the directory would be `/etc/openhab2/automation/jsr223`.
+
+At system start time the scripts will be loaded in an order based on their name and then top-down through the directory hierarchy. Note that while scripts may be placed in subdirectories of `jsr223`, they should not have the same names as scripts in other directories (this may be changed in the future). For example, with the following directory structure...
+
+```
+automation/jsr223
+  01/
+    script1.py
+  00script.py
+  script2.py
+```
+
+the load order will be: `00script.py`, `01/script1.py`, `script2.py`.
```

Comment:
I've added the example. I think the current file loading is very confusing and should be improved. Fortunately, most users will just have scripts in the top-level directory. I need to study the source and do some experimentation to be 100% confident that my current description is fully accurate.